.github/workflows/github-ci.yml

```diff
@@ -0,0 +1,69 @@
+name: Java CI with Maven and Release Artifact
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  release:
+    types: [ released ]
+
+permissions:
+  contents: write
+  actions: read
+  checks: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    outputs:
+      version: ${{ steps.extract_version.outputs.version }}
+      artifactId: ${{ steps.extract_artifactId.outputs.artifactId }}
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          java-version: '23'
+          distribution: 'temurin'
+          cache: maven
+
+      - name: Build with Maven
+        run: mvn -B clean install
```

Comment:
To be double-checked: do we need `mvn install` here (which usually means putting the built jar file into the "local maven repo", not sure if that makes sense here)? Or would  `mvn verify` suffice?